x-pack/plugins/security_solution/public/resolver/store/middleware/current_related_event_fetcher.ts

```diff
@@ -34,28 +34,35 @@ export function CurrentRelatedEventFetcher(
 
     const oldParams = last;
     last = newParams;
+    const newID = selectors.refreshCount(state);
 
     // If the panel view params have changed and the current panel view is the `eventDetail`, then fetch the event details for that eventID.
-    if (!isEqual(newParams, oldParams) && newParams.panelView === 'eventDetail') {
-      const currentEventID = newParams.panelParameters.eventID;
-
+    if (
+      (!isEqual(newParams, oldParams) && newParams.panelView === 'eventDetail') ||
+      (selectors.currentRelatedEventIsStale(state) && newParams.panelView === 'eventDetail')
```

Comment:
not a biggie, but you can probably add the `newParams.panelView === 'eventDetail` logic to the `currentRelatedEventIsStale` selector since I figure we'll only care about that when we're explicitly on that panel